python/ray/dashboard/client/src/api.ts

```diff
@@ -217,14 +217,22 @@ export interface TuneTrial {
   training_iteration: number;
   start_time: string;
   status: string;
-  trial_id: string;
+  trial_id: string | number;
```

Comment:
can we comment on why this change is needed?

Reply:
There was a change to how the trial ID's were being set such that they can now be strings or numbers. Previously, I was casting all of them to strings in the backend; however, this messes with the sorting for this attribute in the Table as it sorts them by their string value("10" < "2") rather than their numerical value.  This tweak is accompanied by a change in the backend that preserves the true type of the trial ID.